### ISSUE TITLE
OpenBSD and FreeBSD?

### ISSUE THREAD

**Issue by standenboy** [pystardust/ani-cli#1136]
So i've recently started to use openbsd on my laptop, and i would have assumed that a shell script would port over to openbsd well, but maybe i'm mistaken

i try to run the script with `ani-cli death note` and it then returns with `checking dependencies` and then the program stops and i'm dropped back into my shell

i'm assuming this means there is a section of code that needs to be re-writen to support the BSD's however i'm not knowledgeable enough to know which section the issue is coming from

if i've missed a setting somewhere or an option i'm sorry for the inconvenience 

if its of any help i'm running openbsd 7.3
i'm using the latest version of ani-cli

if any more info is needed just ask and ill provide everything i can

**Comment by 71zenith** [comment]
Can u send the output of `sh -x $(which ani-cli) death note`

**Comment by standenboy** [comment]
+ version_number=4.3.1
+ agent=Mozilla/5.0 (Windows NT 6.1; Win64; rv:109.0) Gecko/20100101 Firefox/109.0
+ allanime_base=allanime.to
+ mode=sub
+ download_dir=.
+ quality=best
+ no_menu=0
+ uname -a
+ player_function=mpv
+ player_function=mpv
+ use_external_menu=0
+ [ -t 0 ]
+ [ 0 = 0 ]
+ multi_selection_flag=-m
+ [ 0 = 1 ]
+ cache_dir=/home/standenboy/.cache/ani-cli
+ [ ! -d /home/standenboy/.cache/ani-cli ]
+ hist_dir=/home/standenboy/.local/state/ani-cli
+ [ ! -d /home/standenboy/.local/state/ani-cli ]
+ histfile=/home/standenboy/.local/state/ani-cli/ani-hsts
+ [ ! -f /home/standenboy/.local/state/ani-cli/ani-hsts ]
+ search=scrape
+ [ 1 -gt 0 ]
+ printf %s  deathnote
+ sed s|^ ||;s| |+|g
+ query=deathnote
+ shift
+ [ 0 -gt 0 ]
+ printf \33[2K\r\033[1;34mChecking dependencies...\033[0m\n
Checking dependencies...
+ dep_ch curl sed grep fzf
+ dep_ch mpv
+ [ -z deathnote ]
+ printf %s deathnote
+ sed s| |+|g
+ query=deathnote
+ search_anime deathnote
+ anime_list=
+ [ -z  ]
+ die No results found!
No results found!

**Comment by standenboy** [comment]
+ version_number=4.3.1
+ agent=Mozilla/5.0 (Windows NT 6.1; Win64; rv:109.0) Gecko/20100101 Firefox/109.0
+ allanime_base=allanime.to
+ mode=sub
+ download_dir=.
+ quality=best
+ no_menu=0
+ uname -a
+ player_function=mpv
+ player_function=mpv
+ use_external_menu=0
+ [ -t 0 ]
+ [ 0 = 0 ]
+ multi_selection_flag=-m
+ [ 0 = 1 ]
+ cache_dir=/home/standenboy/.cache/ani-cli
+ [ ! -d /home/standenboy/.cache/ani-cli ]
+ hist_dir=/home/standenboy/.local/state/ani-cli
+ [ ! -d /home/standenboy/.local/state/ani-cli ]
+ histfile=/home/standenboy/.local/state/ani-cli/ani-hsts
+ [ ! -f /home/standenboy/.local/state/ani-cli/ani-hsts ]
+ search=scrape
+ [ 2 -gt 0 ]
+ printf %s  death
+ sed s|^ ||;s| |+|g
+ query=death
+ shift
+ [ 1 -gt 0 ]
+ printf %s death note
+ sed s|^ ||;s| |+|g
+ query=death+note
+ shift
+ [ 0 -gt 0 ]
+ printf \33[2K\r\033[1;34mChecking dependencies...\033[0m\n
Checking dependencies...
+ dep_ch curl sed grep fzf
+ dep_ch mpv
+ [ -z death+note ]
+ printf %s death+note
+ sed s| |+|g
+ query=death+note
+ search_anime death+note
+ anime_list=RezHft5pjutwWcE3BtDeath Note (37 episodes)
+ [ -z RezHft5pjutwWcE3BtDeath Note (37 episodes) ]
+ [  -eq  ]
+ 2> /dev/null
+ printf %s RezHft5pjutwWcE3BtDeath Note (37 episodes)
+ sed -n p
+ result=RezHft5pjutwWcE3BtDeath Note (37 episodes)
+ [ -z  ]
+ printf %s RezHft5pjutwWcE3BtDeath Note (37 episodes)
+ nl -w 1
+ nth Select anime:
+ result=RezHft5pjutwWcE3BtDeath Note (37 episodes)
+ [ -z RezHft5pjutwWcE3BtDeath Note (37 episodes) ]
+ printf %s RezHft5pjutwWcE3BtDeath Note (37 episodes)
+ cut -f2
+ title=RezHft5pjutwWcE3BtDeath Note (37 episodes)
+ printf %s RezHft5pjutwWcE3BtDeath Note (37 episodes)
+ tr -d [:punct:]
+ cut -d( -f1
+ tr A-Z  a-z-
+ allanime_title=rezhft5pjutwwce3btdeath-note-
+ printf %s RezHft5pjutwWcE3BtDeath Note (37 episodes)
+ cut -f1
+ id=RezHft5pjutwWcE3BtDeath Note (37 episodes)
+ episodes_list RezHft5pjutwWcE3BtDeath Note (37 episodes)
+ ep_list=
+ [ -z  ]
+ printf %s
+ nth Select episode:  -m
+ ep_no=
+ [ -z  ]
+ exit 1

**Comment by standenboy** [comment]
> It's cut again, isn't it

cut is used multiple times, could you elaborate on what you mean by this

**Comment by port19x** [comment]
cut is a core utility, typically provided by the gnu coreutils. We leverage it a lot, but it has some inconsistencies across the plattforms

**Comment by standenboy** [comment]
Yes of course, do with have any clue as to which cut command is causing the issue?

**Comment by standenboy** [comment]
I am aware of the difference between the two, however I didn't know the gnu utils were available, I'll have to try them out and see if I can get it to work

**Comment by justchokingaround** [comment]
this just looks like a duplicate of issue #1141 
pr #1144 fixes it

**Comment by port19x** [comment]
Assuming it's fixed by #1144. Closing
